gobblin-core/src/main/java/org/apache/gobblin/publisher/HiveRegistrationPublisher.java

```diff
@@ -189,9 +200,27 @@ public void publishMetadata(Collection<? extends WorkUnitState> states) throws I
     // Nothing to do
   }
 
-  private static void addRuntimeHiveRegistrationProperties(State state) {
-    // Use seconds instead of milliseconds to be consistent with other times stored in hive
-    state.appendToListProp(HiveRegProps.HIVE_TABLE_PARTITION_PROPS,
-        String.format("%s:%d", DATA_PUBLISH_TIME, TimeUnit.SECONDS.convert(System.currentTimeMillis(), TimeUnit.MILLISECONDS)));
+  /**
+   * A wrapper for updating partition record count to {@link #pathToRecordCount}.
+   */
+  private void countUpForPath(String path, State state) {
+    long recordCountInWorkUnit = state.getPropAsLong(ConfigurationKeys.WORK_UNIT_HIGH_WATER_MARK_KEY)
```

Comment:
so record count is basically the diff on watermark ? what if the watermark is date? is that acceptable?